cmd/common.cpp

```diff
@@ -297,24 +301,56 @@ void run_preflight_checklist(NodeSettings& node_settings) {
         } else if (node_settings.chain_config.value().chain_id != node_settings.network_id) {
             throw std::runtime_error("Incompatible network id. Command line expects " +
                                      std::to_string(node_settings.network_id) + "; Database has " +
                                      std::to_string(node_settings.chain_config.value().chain_id));
         }
-        std::string chain_name{" unknown/custom network"};
-        auto chains_map{get_known_chains_map()};
-        for (auto& [name, id] : chains_map) {
-            if (id == node_settings.chain_config.value().chain_id) {
-                chain_name = name;
-                break;
+
+        auto known_chain{lookup_known_chain(node_settings.chain_config->chain_id)};
+        if (known_chain.has_value() && *(known_chain->second) != *(node_settings.chain_config)) {
+            // If loaded config is known we must ensure is up-to-date with hardcoded one
+            // Loop all respective JSON members to find discrepancies
+            auto known_chain_config_json{known_chain->second->to_json()};
+            auto active_chain_config_json{node_settings.chain_config->to_json()};
+            bool new_members_added{false};
+            for (auto& [known_key, known_value] : known_chain_config_json.items()) {
+                if (!active_chain_config_json.contains(known_key)) {
+                    // Is this new key a definition of a new fork block or a bomb delay block ?
+                    // If so we need to check its new value must be **beyond** the highest
+                    // header processed.
+
+                    const std::regex block_pattern(R"(Block$)", std::regex_constants::icase);
+                    if (std::regex_match(known_key, block_pattern)) {
+                        if (known_value.get<uint64_t>() <= header_download_progress) {
+                            throw std::runtime_error("Can't apply new chain config key " + known_key + "with value " +
+                                                     known_value.get<std::string>() +
+                                                     " as the database has already a higher header");
+                        }
+                    }
+
+                    new_members_added = true;
+                    continue;
+                }
+
+                auto active_value{active_chain_config_json[known_key]};
+                if (active_value.type_name() != known_value.type_name() || active_value != known_value) {
+                    throw std::runtime_error(
+                        "Hard-coded chain config has incompatible data types or values with stored chain config");
+                }
+            }
+
+            if (new_members_added) {
+                db::update_chain_config(*tx, *(known_chain->second));
+                tx.commit();
+                node_settings.chain_config = db::read_chain_config(*tx);
```

Comment:
Is there a specific reason to read chain config again after update instead of using `known_chain->second`?